src/consensus.rs

```diff
@@ -491,11 +491,33 @@ impl Consensus {
             return Ok(None);
         }
 
-        if self
-            .try_add_origin()
-            .context("failed to propose origin peer URI to consensus")?
-        {
-            return Ok(Some(1));
+        match self.try_add_origin() {
+            // `try_add_origin` is not applicable:
+            // - either current peer is not an origin peer
+            // - or cluster is already established
+            Ok(false) => (),
+
+            // Successfully proposed origin peer to consensus, return to consensus loop to handle `on_ready`
+            Ok(true) => return Ok(Some(1)),
+
+            // Origin peer is not a leader yet, wait for the next tick and return to consensus loop
+            // to tick Raft node
+            Err(err @ TryAddOriginError::NotLeader) => {
+                log::debug!("{err}");
+
+                let timeout_at = previous_tick + tick_period;
+
+                self.runtime
+                    .block_on(async { tokio::time::sleep_until(timeout_at.into()).await });
```

Comment:
How about just [`std::thread::sleep`](https://doc.rust-lang.org/std/thread/fn.sleep.html), isn't that the same here?

Reply:
Good point. I guess, I was not thinking much here, just copied stuff from `recv_update`. 🤦‍♀️

Reply:
Fixed. (Thanks!)